Xamarin.Forms.Xaml.UnitTests/Issues/Gh3862.xaml.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using Xamarin.Forms;
+using Xamarin.Forms.Core.UnitTests;
+
+namespace Xamarin.Forms.Xaml.UnitTests
+{
+	public partial class Gh3862 : ContentPage
+	{
+		public Gh3862()
+		{
+			InitializeComponent();
+		}
+
+		public Gh3862(bool useCompiledXaml)
+		{
+			//this stub will be replaced at compile time
+		}
+
+		[TestFixture]
+		class Tests
+		{
+			[SetUp]
+			public void Setup()
+			{
+				Device.PlatformServices = new MockPlatformServices();
+			}
+
+			[TearDown]
+			public void TearDown()
+			{
+				Device.PlatformServices = null;
+			}
+
+			[TestCase(false), TestCase(true)]
+			public void OnPlatformMarkupInStyle(bool useCompiledXaml)
+			{
+				Device.PlatformServices = new MockPlatformServices { RuntimePlatform = Device.iOS };
```

Comment:
Sadly, `MockPlatformServices` does not currently offer a way (that I can see, anyway) of mocking a idiom. One would have to be added to test the idiom code path...